homeassistant/components/apple_tv/__init__.py

```diff
@@ -1,273 +1,363 @@
+            ]
+        )
+    )
+    if unload_ok:
+        manager = hass.data[DOMAIN].pop(entry.unique_id)
+        await manager.disconnect()
+
+    return unload_ok
+
+
+class AppleTVEntity(Entity):
+    """Device that sends commands to an Apple TV."""
+
+    def __init__(self, name, identifier, manager):
+        """Initialize device."""
+        self.atv = None
+        self.manager = manager
+        self._name = name
+        self._identifier = identifier
+
+    async def async_added_to_hass(self):
+        """Handle when an entity is about to be added to Home Assistant."""
+
+        @callback
+        def _async_connected(atv):
+            """Handle that a connection was made to a device."""
+            self.atv = atv
+            self.async_device_connected(atv)
+            self.async_write_ha_state()
+
+        @callback
+        def _async_disconnected():
+            """Handle that a connection to a device was lost."""
+            self.async_device_disconnected()
+            self.atv = None
+            self.async_write_ha_state()
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, f"{SIGNAL_CONNECTED}_{self._identifier}", _async_connected
             )
-        except DeviceAuthenticationError as ex:
-            hass.components.persistent_notification.async_create(
-                f"Authentication failed! Did you enter correct PIN?<br /><br />Details: {ex}",
-                title=NOTIFICATION_AUTH_TITLE,
-                notification_id=NOTIFICATION_AUTH_ID,
+        )
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{SIGNAL_DISCONNECTED}_{self._identifier}",
+                _async_disconnected,
             )
+        )
 
-        hass.async_add_job(configurator.request_done, instance)
+    def async_device_connected(self, atv):
+        """Handle when connection is made to device."""
 
-    instance = configurator.request_config(
-        "Apple TV Authentication",
-        configuration_callback,
-        description="Please enter PIN code shown on screen.",
-        submit_caption="Confirm",
-        fields=[{"id": "pin", "name": "PIN Code", "type": "password"}],
-    )
+    def async_device_disconnected(self):
+        """Handle when connection was lost to device."""
 
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self._identifier)},
+            "manufacturer": "Apple",
+            "name": self.name,
+        }
 
-async def scan_apple_tvs(hass):
-    """Scan for devices and present a notification of the ones found."""
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
 
-    atvs = await scan_for_apple_tvs(hass.loop, timeout=3)
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._identifier
```

Comment:
Unique ids must be both unique and stable:
https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements

Reply:
See earlier response.